samples/openapi3/client/petstore/python-experimental/tests/test_discard_unknown_properties.py

```diff
@@ -0,0 +1,129 @@
+        api_client = petstore_api.ApiClient(config)
+        data = {
+            'dog': {
+                "class_name": "Dog",
+                "color": "black",
+                "breed": "husky",
+                "unknown_property": "a-value",
+                "more-unknown": [
+                    "a"
+                ]
+            }
+        }
+        # The 'unknown_property' is undeclared, which would normally raise an exception, but
+        # when discard_unknown_keys is set to True, the unknown properties are discarded.
+        response = MockResponse(data=json.dumps(data))
+        deserialized = api_client.deserialize(response,
+            ({str: (petstore_api.Dog,)},), True)
+        self.assertTrue(isinstance(deserialized, dict))
+        self.assertTrue(isinstance(deserialized['dog'], petstore_api.Dog))
+
+    def test_deserialize_cat_do_not_discard_unknown_properties(self):
+        """ deserialize str, Cat) with unknown properties, strict validation is enabled """
+        config = Configuration(discard_unknown_keys=False)
+        api_client = petstore_api.ApiClient(config)
+        data = {
+            'cat': {
+                "class_name": "Cat",
+                "color": "black",
+                "declawed": True,
+                "dynamic-property": 12345,
+            }
+        }
+        response = MockResponse(data=json.dumps(data))
+
+        # Deserializing with strict validation does not raise an exception because the even though
+        # the 'dynamic-property' is undeclared, the 'Cat' schema defines the additionalProperties
+        # attribute.
+        deserialized = api_client.deserialize(response,
+            ({str: (petstore_api.Cat,)},), True)
+        self.assertTrue(isinstance(deserialized, dict))
+        self.assertTrue(isinstance(deserialized['cat'], petstore_api.Cat))
+        self.assertEqual(deserialized['cat']['color'], 'black')
+
+    def test_deserialize_cat_discard_unknown_properties(self):
+        """ deserialize str, Cat) with unknown properties, discard unknown properties """
+        config = Configuration(discard_unknown_keys=True)
+        api_client = petstore_api.ApiClient(config)
+        data = {
+            'cat': {
+                "class_name": "Cat",
+                "color": "black",
+                "declawed": True,
+                "unknown_property": "a-value",
+                "more-unknown": [
+                    "a"
+                ]
+            }
+        }
+        # The 'unknown_property' is undeclared, which would normally raise an exception, but
+        # when discard_unknown_keys is set to True, the unknown properties are discarded.
+        response = MockResponse(data=json.dumps(data))
+        deserialized = api_client.deserialize(response,
+            ({str: (petstore_api.Cat,)},), True)
+        self.assertTrue(isinstance(deserialized, dict))
+        self.assertTrue(isinstance(deserialized['cat'], petstore_api.Cat))
```

Comment:
Can you add a test to check the value of deserialized['cat'].to_dict?
That would prove that the unknown properties have been discarded.

Reply:
done